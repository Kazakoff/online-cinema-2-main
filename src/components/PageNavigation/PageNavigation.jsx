import React from 'react';

export const PageNavigation = (props) => {

  const {
    handlePrevPage,
    page,
    handleNextPage
  } = props

  return (
    <div className='d-flex align-items-center justify-content-center'>

      <button className='btn whiteBtn' style={{ marginRight: '7px' }} onClick={handlePrevPage}>{'<'}</button>

      {page}

      <button className='btn whiteBtn' style={{ marginLeft: '7px' }} onClick={handleNextPage}>{'>'}</button>

    </div>
  )
}
