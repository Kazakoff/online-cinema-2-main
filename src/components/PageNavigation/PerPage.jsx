import React, { useRef, useEffect } from 'react';
import { DropDown } from '../DropDown';

export const PerPage = (props) => {

  const {
    limitArray,
    setLimit,
    limit,
    initialLimit
  } = props

  useEffect(() => {
    setLimit(initialLimit ? initialLimit : limitArray[0])
  }, [])

  return (
    <DropDown style={{ marginLeft: '7px' }} label={limit}>
      {
        limitArray.map(item => {
          return (
            <div
              key={item}
              className='d-flex justify-content-center w100'
              style={limit === item ? { backgroundColor: '#ccc', borderRadius: '100%' } : { cursor: 'pointer' }}
              onClick={() => setLimit(item)} >
              {item}
            </div>
          )
        })
      }
    </DropDown>
  )
}
