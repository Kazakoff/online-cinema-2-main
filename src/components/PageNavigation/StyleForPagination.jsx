import React from 'react';

export const StyleForPagination = (props) => {

  return (
    <div className='d-flex justify-content-center' style={{ marginBottom: '12px' }}>
      {props.children}
    </div>
  )
}
