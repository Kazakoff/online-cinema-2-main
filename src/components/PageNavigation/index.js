export { PageNavigation } from './PageNavigation'
export { PerPage } from './PerPage'
export { StyleForPagination } from './StyleForPagination'