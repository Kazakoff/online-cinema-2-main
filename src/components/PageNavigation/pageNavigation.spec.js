import React from 'react'
import { PageNavigation } from './PageNavigation'

const setUp = () => shallow(<PageNavigation />);

describe('PageNavigation component', () => {
    it('should render PageNavigation component', () => {
        const component = setUp()
        expect(component.debug()).toMatchSnapshot()
    })
})