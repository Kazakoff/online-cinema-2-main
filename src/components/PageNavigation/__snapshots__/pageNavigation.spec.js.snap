// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNavigation component should render PageNavigation component 1`] = `
"<div className=\\"d-flex align-items-center justify-content-center\\">
  <button className=\\"btn whiteBtn\\" style={{...}} onClick={[undefined]}>
    &lt;
  </button>
  <button className=\\"btn whiteBtn\\" style={{...}} onClick={[undefined]}>
    &gt;
  </button>
</div>"
`;

exports[`SignIn component should render SignIn component 1`] = `
"<div className=\\"d-flex align-items-center justify-content-center\\">
  <button className=\\"btn whiteBtn\\" style={{...}} onClick={[undefined]}>
    &lt;
  </button>
  <button className=\\"btn whiteBtn\\" style={{...}} onClick={[undefined]}>
    &gt;
  </button>
</div>"
`;
